### PR TITLE
Exported Plugin utility functions and the craco config to remove a webpack plugin  Issue#159

### PR DESCRIPTION
**Pull request by alexasselin008**
This PR adds the following code :
- Craco now exports the new plugin utility functions created in [this PR](https://github.com/gsoft-inc/craco/pull/219).
- Added documentation for those utility functions.
- Added a way in the craco config to exclude some plugins from webpack. Details in [this issue](https://github.com/gsoft-inc/craco/issues/159)

### CODE REVIEW

File: packages/craco/README.md [hunk 1]
```diff
@@ -165,6 +165,10 @@ module.exports = {
     webpack: {
         alias: {},
         plugins: [],
+        removePlugins: {
+		    pluginNames: [],
+        }
+        removePlugins: (plugins, { env, paths }) => { return plugins; };
```
Comment (patricklafrance) [inline packages/craco/README.md:171]: I would prefer to use the `configure` function for this.

File: packages/craco/README.md [hunk 2]
```diff
@@ -673,10 +677,13 @@ A few utility functions are provided by CRACO to help you develop a plugin:
  - `addBeforeLoaders`
  - `addAfterLoader`
  - `addAfterLoaders`
+ - `getPlugin`
+ - `removePlugins`
+ - `addPlugins`
```
Comment (patricklafrance) [inline packages/craco/README.md:682]: A replacePlugin function is missing.

File: packages/craco/README.md [hunk 3]
```diff
@@ -165,6 +165,10 @@ module.exports = {
     webpack: {
         alias: {},
         plugins: [],
+        removePlugins: {
```
Comment (patricklafrance) [inline packages/craco/README.md:168]: I wasn't thinking about introducing a new prop for this. My thought was to make the plugin prop smarter, by also accepting an array for replacing an existing plugin or removing one.

for exemple:

```
webpack: {
    plugins: [
       new ConfigWebpackPlugin(),
       [new HtmlWebpackPlugin(...), { action: "replace" }],
       ["ReactRefreshWebpackPlugin", { action: "remove" }]
    ]
}
```

While writing this I find it quite sloppy hehe so I am open to something else.

Maybe:

```
webpack: {
    addPlugins: [new ConfigWebpackPlugin()],
    replacePlugins: [{ name: "HtmlWebpackPlugin", new HtmlWebpackPlugin(...) }],
    removePlugins: ["ReactRefreshWebpackPlugin"]
}
```

For `replacePlugins`, the plugin name could maybe be inferred from the new instance.

I might prefer the following style:

```
webpack: {
    plugins: {
       add: [new ConfigWebpackPlugin()],
       replace: [{ name: "HtmlWebpackPlugin", new HtmlWebpackPlugin(...) }],
       remove: ["ReactRefreshWebpackPlugin"]
    }
}
```

What do you think?

File: packages/craco/README.md (hunk 2 above)
Comment (alexasselin008) [reply packages/craco/README.md:681]: The remove and add functions should be enough. A replace function would make it confusing in my opinion

**Comment by alexasselin008** [comment]
Release Notes : 

- New utility methods available :  getPlugin, pluginByName, addPlugins, removePlugins.
- Webpack config override now allow the removal of webpack plugins.

**Pull request merged**